### PR TITLE
OSDOCS-11980: Migrate Nodes to ROSA HCP

### PR DESCRIPTION
**Pull request by bhardesty**
This adds the Nodes book to the new ROSA HCP distro. 

**Note for peer review:** This PR does not add any new content. It just applies conditionalization so that the Nodes book can be included in the ROSA HCP distro.

Version(s):
- enterprise-4.17+

Issue:
https://issues.redhat.com/browse/OSDOCS-11980

Link to docs preview:
https://84462--ocpdocs-pr.netlify.app/openshift-rosa-hcp/latest/nodes/

QE review:
- [x] QE has approved this change. (Approved in #83099.)

### CODE REVIEW

**Review by adellape — COMMENTED** [review]
I did not find any issues with the conditionals.

File: _topic_maps/_topic_map_rosa_hcp.yml [hunk 1]
```diff
@@ -381,3 +379,336 @@ Distros: openshift-rosa-hcp
...
+#   - Name: Using remote worker node at the network edge
+#     File: nodes-edge-remote-workers
+#  Not supported per Michael McNeill
+# - Name: Worker nodes for single-node OpenShift clusters
+#   Dir: nodes
+#   Distros: openshift-rosa-hcp
+#   Topics:
+#   - Name: Adding worker nodes to single-node OpenShift clusters
+#     File: nodes-sno-worker-nodes
+- Name: Using Jobs and DaemonSets
```
[the remark this answers is not in the document]
Comment (bhardesty) [reply _topic_maps/_topic_map_rosa_hcp.yml]: Done - I changed this for OCP, ROSA, ROSA with HCP, and OSD.

**Pull request merged**